### ISSUE TITLE
Tokio 1.0?

### ISSUE THREAD

**Issue by DimiDimit** [rmanoka/async-scoped#5]
It seems that changing `Cargo.toml` "just works", but I haven't tested it much.

**Comment by rmanoka** [comment]
Seems fair;  should do that in the next release.  BTW, do you know if tokio supports any APIs to check if the current thread is a worker thread of the thread-pool?  The current support for tokio panics if [`block_in_place`](https://tokio-rs.github.io/tokio/doc/tokio/task/fn.block_in_place.html) is called outside the thread-pool.

**Comment by DimiDimit** [comment]
I had a look at both the documentation and the source code, but the only things I found are `pub(crate)`...

For reference, here's how `tokio` checks it in [`worker.rs`](https://tokio-rs.github.io/tokio/doc/src/tokio/runtime/thread_pool/worker.rs.html#205-233):
```rust
// Tracks thread-local state
scoped_thread_local!(static CURRENT: Context);

// [...]

pub(crate) fn block_in_place<F, R>(f: F) -> R
where
    F: FnOnce() -> R,
{
    // [...]

    CURRENT.with(|maybe_cx| {
        match (crate::runtime::enter::context(), maybe_cx.is_some()) {
            (EnterContext::Entered { .. }, true) => {
                // We are on a thread pool runtime thread, so we just need to set up blocking.
                // [...]
            }
            (EnterContext::Entered { allow_blocking }, false) => {
                // We are on an executor, but _not_ on the thread pool.
                // That is _only_ okay if we are in a thread pool runtime's block_on method:
                if allow_blocking {
                    // [...]
                } else {
                    // This probably means we are on the basic_scheduler or in a LocalSet,
                    // where it is _not_ okay to block.
                    panic!("can call blocking only when running on the multi-threaded runtime");
                }
            }
            (EnterContext::NotEntered, true) => {
                // This is a nested call to block_in_place (we already exited).
                // [...]
            }
            (EnterContext::NotEntered, false) => {
                // We are outside of the tokio runtime, so blocking is fine.
                // [...]
            }
        }

        // [...]
    });

    // [...]
}
```

**Comment by rmanoka** [comment]
Thanks for looking into this @DimiDimit ; I'll try asking the tokio authors if they could support something like that.  Meanwhile, 0.6.0 has been released , and depends on tokio 1.0
